elastic/kibana
# [ILM] Hide node allocation notices on Cloud

## Summary

With the GA of autoscaling, we are able to simplify notifications regarding the availability of nodes with specific roles in the cluster. Specifically, we don't want to notify users that some nodes are missing if they are using the recommended node allocation. The following notifications should no longer show on cloud:

<img width="1045" alt="Screenshot 2021-02-23 at 09 54 06" src="https://user-images.githubusercontent.com/8155004/108821104-6c3ae580-75bd-11eb-8e93-d6caef61b6bb.png">
<img width="1025" alt="Screenshot 2021-02-23 at 09 54 15" src="https://user-images.githubusercontent.com/8155004/108821113-6d6c1280-75bd-11eb-8f78-ea3ad0ae7ba3.png">
<img width="1044" alt="Screenshot 2021-02-23 at 09 55 55" src="https://user-images.githubusercontent.com/8155004/108821117-6e9d3f80-75bd-11eb-9365-5d394bfd45a1.png">

## How to test

Edit your `kibana.dev.yml` to have the following config to simulate being on cloud:

```
xpack.cloud.id: 'my-cloud-id'
xpack.cloud.deploymentUrl: 'http://fake.url'
```

1. Start kibana `yarn start` (basic license is fine), start ES without specifying node roles for your node --  no `-E node.roles=...`
2. Go to the index lifecycle management UI in the management section
3. Create a new policy and activate both warm and cold phases
4. Expanding the advanced section in both of the above phases and check that there are no notifications under "Data allocation" field:

<img width="1034" alt="Screenshot 2021-03-15 at 15 57 50" src="https://user-images.githubusercontent.com/8155004/111173775-34650380-85a7-11eb-834c-3767d8a654e4.png">


### Checklist

Delete any items that are not applicable to this PR.
